fuuk/people/migrations/0004_refactor_grants_data_migration.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def move_author_person_to_grant(apps, schema_editor):
+    # We can't import the Person model directly as it may be a newer
+    # version than this migration expects. We use the historical version.
+    Grant = apps.get_model("people", "Grant")
+    Institution = apps.get_model("people", "Institution")
+    for grant in Grant.objects.all():
+        grant.investigator_prefix = grant.author.prefix
+        grant.investigator_first_name = grant.author.first_name
+        grant.investigator_last_name = grant.author.last_name
+        grant.investigator_suffix = grant.author.suffix
+        if grant.author.human:
+            human = grant.author.human
+            grant.investigator_human = human
+
+        places = grant.author.place
+        if places.exists() and not grant.author.type:
+            place = places.order_by('id')[0]
+            institutions = Institution.objects.filter(name__icontains=place.name)
+            if not institutions.exists():
+                institution = Institution(name=place.name)
+                institution.name_cs = place.name_cs
+                institution.name_en = place.name_en
+                institution.save()
+                grant.investigator_institution = institution
+            else:
+                grant.investigator_institution = institutions.first()
```

Comment:
To migrate places to institutions correctly, it would be better to maintain a map between places and institutions.